boa/src/builtins/math/mod.rs

```diff
@@ -415,14 +508,15 @@ impl Math {
     /// [spec]: https://tc39.es/ecma262/#sec-math.imul
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/imul
     pub(crate) fn imul(_: &JsValue, args: &[JsValue], context: &mut Context) -> JsResult<JsValue> {
-        Ok(match (
-            args.get(0).map(|x| x.to_u32(context)).transpose()?,
-            args.get(1).map(|x| x.to_u32(context)).transpose()?,
-        ) {
-            (Some(x), Some(y)) => x.wrapping_mul(y) as i32,
-            (_, _) => 0,
-        }
-        .into())
+        // 1. Let a be ℝ(? ToUint32(x)).
+        let x = args.get_or_undefined(0).to_u32(context)?;
+
+        // 2. Let b be ℝ(? ToUint32(y)).
+        let y = args.get_or_undefined(1).to_u32(context)?;
+
+        // 3. Let product be (a × b) modulo 232.
```

Comment:
I think you forgot the `^`?
```suggestion
        // 3. Let product be (a × b) modulo 2^32.
```